File: A/yogurt_sale.py
import sys
input = sys.stdin.readline

def solve():
    n, a, b = map(int, input().split())
    min_cost = 0
    if b < 2 * a:
        min_cost += b * (n // 2)
        n = n % 2
    min_cost += n * a
    print(min_cost)

for _ in range(int(input())):
    solve()

File: A/yogurt_sale.c
#include <stdio.h>

void solve() {
    int n, a, b, c = 0;
    scanf("%d %d %d", &n, &a, &b);
    if (b < 2 * a) {
        c += b * (n / 2);
        n = n % 2;
    }
    c += n * a;
    printf("%d\n", c);
}

int main(void) {
    int t;
    scanf("%d", &t);
    while (t--) {
        solve();
    }
    return 0;
}
